src/main/java/mj/chess/Board.java

```diff
@@ -1,82 +1,133 @@
 package mj.chess;
 
 import mj.chess.pieces.Color;
-import mj.chess.pieces.Pawn;
+import mj.chess.pieces.Piece;
+import mj.chess.pieces.RepresentationMap;
+import mj.chess.pieces.Type;
+import mj.chess.utils.StringUtil;
 
 import java.util.ArrayList;
 import java.util.List;
-import java.util.stream.Collectors;
 
 import static java.util.stream.Collectors.joining;
-
+import static mj.chess.utils.StringUtil.appendNewLine;
 
 public class Board {
-    private static final int MAX_SIZE = 8;
+    private static final int FIRST_ROW = 0;
     private static final int SECOND_ROW = 1;
     private static final int SEVENTH_ROW = 6;
-    private final List<Pawn> pawns = new ArrayList<>();
-    private final StringBuilder rowOfBlackPawns = new StringBuilder();
-    private final StringBuilder rowOfWhitePawns = new StringBuilder();
-    private final String[] cellsOfBoard = {
-            "********",
-            "********",
-            "********",
-            "********",
-            "********",
-            "********",
-            "********",
-            "********"
-    };
+    private static final int EIGHTH_ROW = 7;
+    private static final int SIZE_OF_ROW = 8;
+    private final List<Piece> whitePieces = new ArrayList<>();
+    private final List<Piece> blackPieces = new ArrayList<>();
+    private final String[] cellsOfBoard = new String[SIZE_OF_ROW];
 
     public Board() {
     }
 
-    public void add(Pawn pawn) {
-        pawns.add(pawn);
+    private void addPiece(Piece piece) {
+        if (piece.isWhite()) {
+            whitePieces.add(piece);
+        }
+        if (piece.isBlack()) {
+            blackPieces.add(piece);
+        }
```

Comment:
`Piece` 는 흑이거나 백일 거라고 생각됩니다. 또 앞으로의 요구사항에도 거의 100%의 확률로 색깔이 추가될 여지는 없어 보여요. 
그렇다면 아래와 같이 간소화할 수 있을 것 같네요. 

```suggestion
        if (piece.isWhite()) {
            whitePieces.add(piece);
        }
        blackPieces.add(piece);

Reply:
'변경사항에 대한 여지'를 모든 코드에 일일이 반영하는 것은 전체 코드의 간결성을 크게 훼손하는군요!  

wheejuni의 포인트는 서비스의 변경사항이 발생해도, 코어한 일부 코드의 변경만으로 프로그램 전체에 서비스 변경사항이 반영되어야 한다는 것으로 이해했습니다.
특히 '설계적 이점'이라는 키워드를 계속 강조하신 이유가 설계 단계에서 이러한 부분이 반영되어야 한다는 것이구요.